File: backend/app/schemas/disease.py
"""
Disease Detection Schemas

Module: Phase 1 → Module 8 → Disease Detection
Author: VerdiGO Backend Team
"""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from app.enums.disease_severity import DiseaseSeverityEnum


class DiseaseDetectionResponseSchema(BaseModel):
    id: UUID
    farm_id: UUID

    image_url: str
    crop_type: str | None = None

    is_healthy: bool
    disease_name: str | None = None
    confidence: float
    severity: DiseaseSeverityEnum

    treatment: list[str]
    prevention_tips: list[str]

    ai_provider: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class DiseaseHistoryResponseSchema(BaseModel):
    farm_id: UUID
    detections: list[DiseaseDetectionResponseSchema]

File: frontend/src/types/disease.ts
export type DiseaseSeverity = "none" | "low" | "moderate" | "high" | "critical";

export interface DiseaseDetection {
  id: string;
  farm_id: string;
  image_url: string;
  crop_type: string | null;
  is_healthy: boolean;
  disease_name: string | null;
  confidence: number;
  severity: DiseaseSeverity;
  treatment: string[];
  prevention_tips: string[];
  ai_provider: string;
  created_at: string;
}

export interface DiseaseHistoryResponse {
  farm_id: string;
  detections: DiseaseDetection[];
}